### PR TITLE
add hooks parameter to vernier

### PR DESCRIPTION
**Pull request by lHydra**
### What is the purpose of this pull request?

Add hooks configuration parameter for `TestProf::Vernier` to add event markers from Active Support Notifications

### Checklist

- [x] I've added tests for this change
- [x] I've added a Changelog entry
- [x] I've updated a documentation

Closes https://github.com/test-prof/test-prof/issues/289

### CODE REVIEW

File: lib/test_prof/vernier.rb [hunk 1]
```diff
@@ -19,14 +19,15 @@ module TestProf
...
-      attr_accessor :mode, :target, :interval
+      attr_accessor :mode, :target, :interval, :hooks
 
       def initialize
         @mode = ENV.fetch("TEST_VERNIER_MODE", :wall).to_sym
         @target = (ENV["TEST_VERNIER"] == "boot") ? :boot : :suite
 
         sample_interval = ENV["TEST_VERNIER_INTERVAL"].to_i
         @interval = (sample_interval > 0) ? sample_interval : nil
+        @hooks = (ENV["TEST_VERNIER_HOOKS"] == "rails") ? :rails : nil
```
Comment (palkan) [inline lib/test_prof/vernier.rb:30]: We should probably take into account possible future named hooks; so, having smith like: `ENV["TEST_VERNIER_HOOKS"]&.split(",").map(&:to_sym)` would be better.

**Review by palkan — CHANGES_REQUESTED** [review]
Thanks!

Let's try to add a test? At least a smoke test (simply checking that when the hook is provided we do not fail). We can also try to parse the report file and check if it contains some ActiveSupport event; though not sure if it's doable.

File: lib/test_prof/vernier.rb (hunk 1 above)
Comment (lHydra) [reply lib/test_prof/vernier.rb]: done

**Comment by lHydra** [comment]
> Thanks!
> 
> Let's try to add a test? At least a smoke test (simply checking that when the hook is provided we do not fail). We can also try to parse the report file and check if it contains some ActiveSupport event; though not sure if it's doable.

Report contains `markerSchema`, which includes rails events when the hook is enabled. So in the test, I decided to check for a rails event

**Review by palkan — APPROVED** [review]
🚀

**Pull request merged**